Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/lucene/impl/LuceneDAO.java

```diff
@@ -40,9 +42,8 @@ public void init() {
      * Stores the number of indexed word count along with the timestamp
      */
     public void storeValues(long numberOfIndexedWords, String action) {
-        // Authentication auth =
-        // SecurityContextHolder.getContext().getAuthentication();
-        IndexingEvent bean = new IndexingEvent(new Date(), numberOfIndexedWords, action, "some user");
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
why don't you pass the username. This isn't very clean. Your DAO depends on Spring security this way, which it really shouldn't.